### PR TITLE
Increased specificity of Windows platform check

### PR DESCRIPTION
**Pull request by polizz**
**Summary**
Extended regex to match only Windows. Previous regex was matching Darwin in addition to Windows.

yarn run test > all tests pass.

#2261

### CODE REVIEW

File: src/cli/commands/run.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export async function run(
     for (const action of actions) {
       const cmd = scripts[action];
       if (cmd) {
-        const isWin = /win/.test(process.platform);
+        const isWin = /win32/.test(process.platform);
```
Comment (Daniel15) [inline src/cli/commands/run.js]: Could this just be `process.platform === 'win32'`? Looks like that's what Node.js itself does, so it should be safe: https://github.com/nodejs/node/blob/db1087c9757c31a82c50a1eba368d8cba95b57d0/lib/os.js#L6

Comment (polizz) [inline src/cli/commands/run.js]: Certainly. I noticed another section in execute-lifecycle that does just that.

File: __tests__/commands/_helpers.js [hunk 2]
```diff
@@ -131,3 +131,11 @@ export async function run<T, R>(
     throw new Error(`${err && err.stack} \nConsole output:\n ${out}`);
   }
 }
+export const setPlatform = (platform : string) : string => {
+  const previous = process.platform;
+  Object.defineProperty(process, 'platform', {
```
Comment (Daniel15) [inline __tests__/commands/_helpers.js]: I wish Node.js had a decent dependency injection framework, this is super hacky 😛

Comment (polizz) [inline __tests__/commands/_helpers.js]: I'm going to redo that. It's not effective at all 😰 I originally was pondering just passing platform into **run** so the test could directly send it, but, to your point, that seems corny.

**Pull request merged**